scripts/mbedtls_framework/test_case.py

```diff
@@ -10,12 +10,34 @@
 import sys
 from typing import Iterable, List, Optional
 
+from . import build_tree
+from . import psa_information
 from . import typing_util
 
+HASHES_3_6 = {
+    "PSA_WANT_ALG_MD5" : "MBEDTLS_MD_CAN_MD5",
+    "PSA_WANT_ALG_RIPEMD160" : "MBEDTLS_MD_CAN_RIPEMD160",
+    "PSA_WANT_ALG_SHA_1" : "MBEDTLS_MD_CAN_SHA1",
+    "PSA_WANT_ALG_SHA_224" : "MBEDTLS_MD_CAN_SHA224",
+    "PSA_WANT_ALG_SHA_256" : "MBEDTLS_MD_CAN_SHA256",
+    "PSA_WANT_ALG_SHA_384" : "MBEDTLS_MD_CAN_SHA384",
+    "PSA_WANT_ALG_SHA_512" : "MBEDTLS_MD_CAN_SHA512",
+    "PSA_WANT_ALG_SHA3_224" : "MBEDTLS_MD_CAN_SHA3_224",
+    "PSA_WANT_ALG_SHA3_256" : "MBEDTLS_MD_CAN_SHA3_256",
+    "PSA_WANT_ALG_SHA3_384" : "MBEDTLS_MD_CAN_SHA3_384",
+    "PSA_WANT_ALG_SHA3_512" : "MBEDTLS_MD_CAN_SHA3_512"
+}
+
+PK_MACROS_3_6 = {
+    "PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY" : "MBEDTLS_PK_HAVE_ECC_KEYS",
+    "PSA_HAVE_ALG_SOME_ECDSA" : "MBEDTLS_PK_CAN_ECDSA_SOME",
```

Comment:
I don't think we need the last three entries. `egrep 'MBEDTLS_PK_CAN_ECDSA_|SA_HAVE_ALG_ECDSA_|PSA_HAVE_ALG_SOME_ECDSA' **/*.py` finds no occurrence. And without them, we can have a simpler scheme where the input is always a PSA alg or key type (`PSA_ALG_xxx` or `PSA_KEY_TYPE_xxx` as opposed to `PSA_WANT_...` or `PSA_HAVE_...`).

(Also, I'm not sure is those are the correct translations.)

Reply:
I included all PK symbols that I replaced (as part of my previous legacy symbol work) event though `PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY` is the only one used in the framework at the moment. Similarly, `PSA_ALG_SHA_256` is the only one used out of the hashes.

Reply:
Ok, makes sense. I'm inclined to keep all the hashes, because handling all of them is no harder than handling just one, but drop the extra PK macros, as they might be trickier.